### PR TITLE
msgfileparser: Bugfix - Replace direct filesystem access with Vault utility functions

### PR DESCRIPTION
**Pull request by sodle-splunk**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [x] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
    - N/A
- [X] If this is a new connector or you are adding new actions
    - N/A
- [x] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
    - This PR fixes code that was not previously compliant with this rule
- [X] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.
    - N/A

## Release Notes (REQUIRED)
* Use SOAR's API to access data in the Vault, instead of directly manipulating the filesystem. [PAPP-32445]
* Replace the unreleased `outlook-msg-file-converter` dependency with the much-better-maintained `msg_parser`
* Update `min_phantom_version` to 6.1.1

## What is the current behavior? (OPTIONAL)
* The action uses direct filesystem access to create and read files in the Vault
* The action uses the `outlook-msg-file-converter` Python package to read Outlook files. This package is not released on PyPI and does not appear to be actively maintained

## What is the new behavior? (OPTIONAL)
* The action uses the Vault API to create and read files in the Vault
* The action uses the `msg_parser` dependency to read Outlook files

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: msgfileparser_connector.py [hunk 1]
```diff
@@ -567,7 +452,7 @@ def _create_container(self, action_result, email_artifact, label, container_seve
         if not container.get('name'):
             container['name'] = os.path.basename(vault_path)
 
-        # we don't/can't add a raw_email to this container, .msg file does not contain the email
+        # we don't/can't add a raw_email to this container, .message file does not contain the email
```
Comment (alexa-phantom) [inline msgfileparser_connector.py]: This one seems like it should stay `.msg` as a file extension

File: msgfileparser_connector.py [hunk 2]
```diff
@@ -602,25 +487,26 @@ def _handle_extract_email(self, param):
             if phantom.is_fail(ret_val):
                 return action_result.set_status(phantom.APP_ERROR, MSGFILEPARSER_CONTAINERS_NOT_FOUND_ERR.format(message))
 
-        # get the .msg file from the vault
+        # get the .message file from the vault
```
Comment (alexa-phantom) [inline msgfileparser_connector.py]: Same

File: msgfileparser_connector.py (hunk 1 above)
Comment (sodle-splunk) [reply msgfileparser_connector.py]: good catch, fixed

File: msgfileparser_connector.py (hunk 2 above)
Comment (sodle-splunk) [reply msgfileparser_connector.py]: fixed

**Review by alexa-phantom — APPROVED** [review]

**Pull request merged**